lib/types/local-time.js

```diff
@@ -27,257 +29,209 @@ const nanoSecInMillis = Long.fromNumber(1000000);
  * @private
  * */
 const millisInDay = 86400000;
-/**
- *
- * Creates a new instance of LocalTime.
- * @class
- * @classdesc A time without a time-zone in the ISO-8601 calendar system, such as 10:30:05.
- * <p>
- *   LocalTime is an immutable date-time object that represents a time, often viewed as hour-minute-second. Time is represented to nanosecond precision. For example, the value "13:45.30.123456789" can be stored in a LocalTime.
- * </p>
- * @param {Long} totalNanoseconds Total nanoseconds since midnight.
- * @constructor
- */
-function LocalTime(totalNanoseconds) {
-    if (!(totalNanoseconds instanceof Long)) {
-        throw new Error("You must specify a Long value as totalNanoseconds");
-    }
-    if (
-        totalNanoseconds.lessThan(Long.ZERO) ||
-        totalNanoseconds.greaterThan(maxNanos)
-    ) {
-        throw new Error("Total nanoseconds out of range");
-    }
-    this.value = totalNanoseconds;
 
+class LocalTime {
     /**
-     * Gets the hour component of the time represented by the current instance, a number from 0 to 23.
-     * @type Number
-     */
-    this.hour = this._getParts()[0];
-    /**
-     * Gets the minute component of the time represented by the current instance, a number from 0 to 59.
-     * @type Number
+     *
+     * Creates a new instance of LocalTime.
+     * @class
+     * @classdesc A time without a time-zone in the ISO-8601 calendar system, such as 10:30:05.
+     * <p>
+     *   LocalTime is an immutable date-time object that represents a time, often viewed as hour-minute-second. Time is represented to nanosecond precision. For example, the value "13:45.30.123456789" can be stored in a LocalTime.
+     * </p>
+     * @param {Long} totalNanoseconds Total nanoseconds since midnight.
+     * @constructor
      */
-    this.minute = this._getParts()[1];
-    /**
-     * Gets the second component of the time represented by the current instance, a number from 0 to 59.
-     * @type Number
-     */
-    this.second = this._getParts()[2];
-    /**
-     * Gets the nanoseconds component of the time represented by the current instance, a number from 0 to 999999999.
-     * @type Number
-     */
-    this.nanosecond = this._getParts()[3];
-}
+    constructor(totalNanoseconds) {
```

Comment:
Maybe consider extending the API by adding the ability to constuct this type from BigInt?

But this can be a separate issue. (@wprzytula @Stapox35)

Reply:
If it was constructible from BigInt, then it still must be.
If it was not, then it's up to you. If you consider such extension QoL, then go on - either here or in an issue.

Reply:
In DataStax, LocalTime has constructor with Long. I created issue for this. 
https://docs.datastax.com/en/developer/nodejs-driver/4.7/api/module.types/class.LocalTime/index.html#local-time